lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -2679,6 +2718,120 @@ instance DecodeAddress t => FromJSON (ApiConstructTransactionData t) where
+               cs -> for cs $ \pair ->
+                   parseAda pair <|> parseTokens pair
+       parseAda (numTxt, num) =
+           if numTxt == "lovelace" then do
+               q <- parseJSON num
+               pure (Just q, Nothing)
+           else
+               fail "expected 'lovelace' key"
+       parseTokens (numTxt, obj) =
+           if numTxt == "lovelace" then
+               fail "expected policyId"
+           else do
+               let processTokensPerPolicyId o =
+                       case HM.toList o of
+                           [] -> fail "tokens should not be empty"
+                           cs -> for (reverse cs) $ \(tNameTxt, tQuantity) -> do
+                               q <- parseJSON tQuantity
+                               let tNameE = W.mkTokenName $ T.encodeUtf8 tNameTxt
+                               case tNameE of
+                                   Right tName -> pure (tName, TokenQuantity q)
+                                   Left _ -> fail "invalid token name"
+               tokenPolicy <- parseJSON (String numTxt)
+               tokenPairs <- withObject "Tokens with given policyId" processTokensPerPolicyId obj
+               pure (Nothing, Just (tokenPolicy, NE.fromList tokenPairs))
+
+instance EncodeAddress n => ToJSON (ApiTxOut n) where
+    toJSON (ApiTxOut addr data' (Quantity amt) (ApiT assets')) = case data' of
+        Nothing -> object objShared
+        Just content ->  object (objShared ++ ["data" .= toJSON content])
+      where
+        objShared =
+            [ "address" .= toJSON addr
+            , "value" .= object (["lovelace" .= toJSON amt] ++ tokens)
+            ]
+        tokenPair (tName, (TokenQuantity quantity)) =
+            [T.decodeLatin1 (W.unTokenName tName) .= toJSON quantity]
+        addEntry policyId tokens' acc = acc ++
+            [ toText policyId .= object (concatMap tokenPair (NE.toList $ NEMap.toList tokens')) ]
+        tokens = Map.foldrWithKey addEntry [] $ toNestedMap assets'
+
+instance DecodeAddress n => FromJSON (ApiExternalInput n) where
+    parseJSON = withObject "ApiExternalInput" $ \o -> do
+        txInVal <- o .: "txIn"
+        txOutVal <- o .: "txOut"
+        ApiExternalInput <$> parseJSON txInVal <*> parseJSON txOutVal
+instance EncodeAddress n => ToJSON (ApiExternalInput n) where
+    toJSON (ApiExternalInput ins outs) = object
+        [ "txIn" .= toJSON ins
+        , "txOut" .= toJSON outs ]
+
+instance DecodeAddress n => FromJSON (ApiBalanceTransactionPostData n) where
+    parseJSON = withObject "ApiBalanceTransactionPostData" $ \o -> do
+        cbor <- o .: "transaction" >>= (\trObj -> trObj .: "cborHex")
+        cosigners <- o .: "signatories"
+        bs <- fmap getApiBytesT $ parseJSON @(ApiBytesT 'Base16 ByteString) cbor
+        let sealedTxFromBytes _x = undefined
+        case sealedTxFromBytes bs of
+            Left err -> fail $ "cborHex seems to be not deserializing correctly due to " -- <> show err
```

Comment:
Commented code ☝️

Reply:
I know, I commented/undefined several places to compile (it was before sign tx PR merged). Now, after sign tx PR it is not the case anymore